src/pages/ReimbursementAccount/ACHContractStep.js

```diff
@@ -115,56 +135,31 @@ class ACHContractStep extends React.Component {
-            BankAccounts.updateReimbursementAccountDraft({beneficialOwners});
-            return {beneficialOwners};
-        });
-
-        // Prepare inputKeys for clearing errors
-        const inputKeys = _.keys(values);
-
-        // dob field has multiple validations/errors, we are handling it temporarily like this.
-        if (_.contains(inputKeys, 'dob')) {
-            inputKeys.push('dobAge');
-        }
-        this.clearErrors(_.map(inputKeys, inputKey => `beneficialOwnersErrors.${ownerIndex}.${inputKey}`));
-    }
-
-    submit() {
-        if (!this.validate()) {
-            return;
-        }
-
+    submit(values) {
         const bankAccountID = lodashGet(store.getReimbursementAccountInSetup(), 'bankAccountID');
 
-        // If they did not select that there are other beneficial owners, then we need to clear out the array here. The
-        // reason we do it here is that if they filled out several beneficial owners, but then toggled the checkbox, we
-        // want the data to remain in the form so we don't lose the user input until they submit the form. This will
-        // prevent the data from being sent to the API
-        this.setState(prevState => ({
-            beneficialOwners: !prevState.hasOtherBeneficialOwners ? [] : prevState.beneficialOwners,
-        }),
-        () => BankAccounts.updateBeneficialOwnersForBankAccount({...this.state, beneficialOwners: JSON.stringify(this.state.beneficialOwners), bankAccountID}));
-    }
+        const beneficialOwners = !this.state.hasOtherBeneficialOwners ? []
+            : _.map(this.state.beneficialOwners, ownerKey => ({
+                firstName: lodashGet(values, `beneficialOwner.${ownerKey}.firstName`),
+                lastName: lodashGet(values, `beneficialOwner.${ownerKey}.lastName`),
+                dob: lodashGet(values, `beneficialOwner.${ownerKey}.dob`),
+                ssnLast4: lodashGet(values, `beneficialOwner.${ownerKey}.ssnLast4`),
+                street: lodashGet(values, `beneficialOwner.${ownerKey}.street`),
+                city: lodashGet(values, `beneficialOwner.${ownerKey}.city`),
+                state: lodashGet(values, `beneficialOwner.${ownerKey}.state`),
+                zipCode: lodashGet(values, `beneficialOwner.${ownerKey}.zipCode`),
+            }));
 
-    /**
-    * @param {Object} fieldName
-    */
-    toggleCheckbox(fieldName) {
-        this.setState((prevState) => {
-            const newState = {[fieldName]: !prevState[fieldName]};
-            BankAccounts.updateReimbursementAccountDraft(newState);
-            return newState;
+        BankAccounts.updateBeneficialOwnersForBankAccount({
+            ownsMoreThan25Percent: values.ownsMoreThan25Percent,
+            hasOtherBeneficialOwners: values.hasOtherBeneficialOwners,
+            acceptTermsAndConditions: values.acceptTermsAndConditions,
+            certifyTrueInformation: values.certifyTrueInformation,
```

Comment:
Are we allow spreading the object? These keys have duplicate keys as the values.